controllers/host/host_controller.go

```diff
@@ -1173,7 +1173,8 @@ func (r *HostReconciler) ReconcileHostByState(
 		}
 
 		// Clean up strategy required after Disabled and enabled attributes are all in-sync
-		if strategy_required {
+		if strategy_required &&
+			!(r.CloudManager.IsWaitingForStrategySent() || r.CloudManager.GetStrategySent()) {
```

Comment:
I think the major problem is this condition: you want to prevent the strategy_required to be overridden to not required too early, which may happen if the strategy was not created yet.

I'm okay with this conditional check. Just expect to enhance the "IsWaitForStrategysent()":

- it actually does not check the strategy status, can we rename  as something like e.g. "GetStrategyExpectedByOtherReoncilers"
- Add debug/v2 level message to indicate the check of the prop
- Update the description above(better have a decision tree) to help to read

- 

1.

Reply:
Done